### ISSUE TITLE
Extra letter added when using  [**]

### ISSUE THREAD

**Issue by fridde** [dannyvankooten/AltoRouter#183]
After adding all my standard routes, I have added `['GET', '[**:url]', 'Error#notFound']` to send any non-matching request to my error handler.
I also set the basePath in the constructor to `myfancywebsite`. 

The weird thing, upon inspecting `$match["params"]["url"]`  for `localhost/myfancywebsite/nonexistingpage` I get the string `e/nonexistingpage`

Any idea where the `e/` could come from? I've been looking everywhere in the source code and have been debugging for a while now. Everything else works just fine.

To summarize
```
$routes = [[some, uninteresting, route,], [some, other, route], ..., [yet, another, route]];
$routes[] = ['GET', '[**:url]', 'Error#notFound'];
$router =  new AltoRouter($routes, 'myfancywebsite');
$match = $router->match();

print_r($match["params"]);

//result for localhost/myfancywebsite/nonexistingpage 
// Array ( [url] => e/nonexistingpage )
```

**Comment by manishbathla** [comment]
Am new to this router myself but after the closure, if there is no match, this closure below will throw a 404 anyways. 

// call closure or throw 404 status
if( $match && is_callable( $match['target'] ) ) {
	call_user_func_array( $match['target'], $match['params'] ); 
} else {
	// no route was matched
	header( $_SERVER["SERVER_PROTOCOL"] . ' 404 Not Found');
}

Why add the extra map? Just curious.

**Comment by fridde** [comment]
It seems like I have found the culprit myself.

The basePath should be prepended with `/`.
```
$router =  new AltoRouter($routes, 'myfancywebsite'); //WRONG
$router =  new AltoRouter($routes, '/myfancywebsite'); // works!
```
I have no idea why the result was so weird. Let's move on!

**Comment by koenpunt** [comment]
Somewhat related to #171
